docs/community/maintainers-guide/releasing.rst

```diff
@@ -0,0 +1,224 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2020 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+Releasing
+=========
+
+In general, you have two types of releases:
+
+- **Major/Minor-level release** - Often these releases add significant new
+  features and needs to go through extensive testing.
+- **Patch-level release** - Often these releases fix security issues or
+  important bugs.
+
+For major/minor-level releases, you'll often be working only with the
+``master``-branch and only make a single PyPI release. For patch-level
+releases you'll often be working with multiple maintenance branches and make
+several PyPI releases for all currently maintained versions of Invenio.
+
+Depending on the package that you are releasing, you will have to follow
+a different procedure. You might have to release:
+
+- one of the Python or JavaScript/React modules (happening rather often)
+- the Invenio framework, the ``invenio`` package (happening less often)
+
+Before you start
+----------------
+Prior to preparing a new Invenio release, please ensure you have the latest
+changes in your local environment for the maintenance branches and ``master``:
+
+1. Check that all relevant PRs have been merged. For the ``invenio`` package,
+   check that all relevant Invenio modules have been released.
+2. Fetch latest changes:
+    - ``git fetch inveniosoftware``
```

Comment:
Minor: I think a maintainer should be more than aware of this... But `inveniosoftware` is a remote, might have another name (i.e. in my case is `upstream` to be compatible with `hub`). Still as said, not sure is worth mentioning?

Reply:
I'm using hub, and I get ``inveniosoftware`` as remote :-) That said, I would include this in a different section on developer environment setup and simply link there.